### ISSUE TITLE
alicloud_disk_attachment implicit dependency didn't work as expected for multiple instances

### ISSUE THREAD

**Issue by yokzy88** [terraform-providers/terraform-provider-alicloud#1402]
Hi,
Running terraform apply for attaching disk together with instance creation, produce the following errors due to incorrect implicit dependencies:

> Error: Error running plan: 1 error occurred:
        * module.web.alicloud_disk_attachment.mongoB_data_attachment: 1 error occurred:
        * module.web.alicloud_disk_attachment.mongoB_data_attachment: Resource 'alicloud_instance.mongo' not found for variable 'alicloud_instance.mongo.1.id'

### Terraform Version
Terraform v0.11.14
+ provider.alicloud v1.50.0

### Affected Resource(s)
Please list the resources as a list, for example:
- instance
- disk

### Terraform Configuration Files
```hcl
resource "alicloud_instance" "mongo" {
  count             = "2"
  security_groups   = ["${var.security_group_id}"]
  instance_type         = "${var.mongo_instance_type}"
  system_disk_category  = "${var.disk_type}"
  image_id              = "${var.mongo_instance_image}"
  vswitch_id            = "${element(var.vswitch_ids, count.index % 2)}"
  security_enhancement_strategy  = "Active"
}
resource "alicloud_disk_attachment" "mongoA_data_attachment" {
  disk_id     = "${alicloud_disk.mongoA_data.id}"
  instance_id = "${alicloud_instance.mongo.0.id}"
  depends_on = [
      "alicloud_instance.mongo"
  ]
}
resource "alicloud_disk_attachment" "mongoB_data_attachment" {
  disk_id     = "${alicloud_disk.mongoB_data.id}"
  instance_id = "${alicloud_instance.mongo.1.id}"
  depends_on = [
      "alicloud_instance.mongo"
  ]
}
```

### Expected Behavior
What should have happened?
 - Terraform apply should success

### Actual Behavior
What actually happened?
 - Terraform apply failed

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Notes that it will work when I put explicit dependency depends_on [ "alicloud_instance.mongo" ]

### References

**Comment by xiaozhu36** [comment]
HI @yokzy88 This is a bug of terrraofrm version `0.11.4` and it has been fixed by terraform `0.12.x` and you can update and check it.

In addition, if you still want to use terraform `0.11.4`, you can use `"${element(alicloud_instance.mongo.*.id, 1)}" ` instead.

**Comment by xiaozhu36** [comment]
Closed by no reply
